src/components/v2/Table/index.tsx

```diff
@@ -41,92 +23,88 @@ export const Table = ({
   rowKeyIndex,
   className,
   gridTemplateColumns,
+  gridTemplateRowsMobile = '1fr',
+  isMobileView,
 }: ITableProps) => {
   const styles = useStyles();
-  const [orderBy, setOrderBy] = useState<typeof columns[number]['key'] | undefined>(
-    initialOrder?.orderBy,
-  );
-  const [orderDirection, setOrderDirection] = useState<'asc' | 'desc' | undefined>(
-    initialOrder?.orderDirection,
-  );
-  const onRequestOrder = (property: typeof columns[number]['key']) => {
-    let newOrder: 'asc' | 'desc' = 'asc';
-    if (property === orderBy) {
-      newOrder = orderDirection === 'asc' ? 'desc' : 'asc';
-    }
-    setOrderBy(property);
-    setOrderDirection(newOrder);
-  };
-
-  const rows = React.useMemo(() => {
-    // order in place
-    if (!orderBy) {
-      return data;
-    }
-    const rowIndex = columns.findIndex(column => column.key === orderBy);
-    const newRows = [...data];
-    newRows.sort((a, b) => {
-      if (+a[rowIndex].value < +b[rowIndex].value) {
-        return orderDirection === 'asc' ? -1 : 1;
-      }
-      if (+a[rowIndex].value > +b[rowIndex].value) {
-        return orderDirection === 'asc' ? 1 : -1;
-      }
-      return 0;
-    });
-    return newRows;
-  }, [data, orderBy, orderDirection]);
-
-  /* gridStyles provides grid-template-columns styles for rows in rows for thead and tbody */
-  const gridStyles =
-    gridTemplateColumns ||
-    useMemo(
-      () =>
-        /* if gridTemplateColumns prop is not passed from parent component, we create similar fractions by default */
-        columns.map(() => '1fr').join(' '),
-      [columns],
-    );
+  const { rows, orderBy, orderDirection, onRequestOrder, gridColumns } = useTable({
+    columns,
+    initialOrder,
+    data,
+    gridTemplateColumns,
+    isMobileView,
+  });
 
   return (
-    <Paper css={styles.root} className={className}>
-      {title && <h4 css={styles.title}>{title}</h4>}
-
-      <TableContainer>
-        <TableMUI css={styles.table({ minWidth: minWidth ?? '0' })} aria-label={title}>
-          <Head
-            columns={columns}
-            orderBy={orderBy}
-            orderDirection={orderDirection}
-            onRequestOrder={onRequestOrder}
-            css={styles.getTemplateColumns({ gridStyles })}
-          />
-
-          {/* TODO: add loading state */}
-
-          {/* TODO: add error state */}
+    <Paper css={[styles.root, isMobileView && styles.rootMobile]} className={className}>
+      {title && <h4 css={[styles.title, isMobileView && styles.titleMobile]}>{title}</h4>}
 
-          <TableBody>
-            {rows.map(row => (
-              <TableRow
-                hover
-                key={row[rowKeyIndex].value.toString()}
-                onClick={e => rowOnClick && rowOnClick(e, row)}
-                css={styles.getTemplateColumns({ gridStyles })}
+      {isMobileView ? (
```

Comment:
We could render both tables and hide one or the other with responsive css

Reply:
This could also be hardcoded because our responsive breakpoints for mobile are static

Reply:
In this case, because we rely on JavaScript anyway to set the grid layout of the table, then I think we using JS to define what layout to display